sdk/include/platform/sunburst/platform-pwm.hh

```diff
@@ -46,23 +30,45 @@ struct SonataPulseWidthModulation
 		 * as only an 8 bit counter is being used.
 		 */
 		uint32_t period;
-	} outputs[OutputCount];
 
-	/*
-	 * Sets the output of a specified pulse-width modulated output.
-	 *
-	 * The first argument is the index of the output. The second argument is
-	 * the period (length) of the output wave represented as a counter of
-	 * system clock cycles. The third argument is the number of clock cycles
-	 * for which a high pulse is sent within that period.
-	 *
-	 * So for example `output_set(0, 200, 31)` should set a 15.5% output.
-	 */
-	void output_set(uint32_t index, uint8_t period, uint8_t dutyCycle) volatile
+		/*
+		 * Sets the output of a specified pulse-width modulated output.
+		 *
+		 * @param period The length of the output wave represented as a counter
+		 * of system clock cycles.
+		 * @param dutyCycle The number of clock cycles for which a high pulse is
+		 * sent within that period.
+		 *
+		 * So for example `output_set(0, 200, 31)` should set a 15.5% output.
+		 * For a constant high output (100% duty cycle), set the dutyCycle >
+		 * period.
+		 */
+		void output_set(uint8_t period, uint8_t dutyCycle) volatile
+		{
+			this->period    = period;
+			this->dutyCycle = dutyCycle;
+		}
+	};
+
+	/// A convenience structure that can map onto multiple PWM outputs.
+	template<size_t NumberOfPwms = 6>
+	struct Array
```

Comment:
Is there a reason that this isn't a `std::array`?  It looks like it's equivalent to a subset of `std:array`.

Reply:
I originally used `std::array` but found a lot of `std::array`'s methods don't work with `utils::NoCopyNoMove` elements.